src/MainMenu.cpp

```diff
@@ -0,0 +1,54 @@
+#include <D3DX8.h>
+#include <windows.h>
+
+#include "MainMenu.hpp"
+#include "Supervisor.hpp"
+
+#pragma optimize("s", on)
+ZunResult MainMenu::BeginStartup()
+{
+    D3DXVECTOR3 vector3Ptr; // we have to add Ptr,
+                            // because otherwise it gets 0.7% less on decomp.me for some reason
+    DWORD time;
+    int i;
+
+    if (LoadTitleAnm(this) != ZUN_SUCCESS)
+    {
+        g_Supervisor.curState = 4;
+        return ZUN_ERROR;
+    }
+    else
+    {
+        if (g_Supervisor.startupTimeBeforeMenuMusic > 0)
+        {
+            time = timeGetTime();
+            while ((time - g_Supervisor.startupTimeBeforeMenuMusic >= 0) &&
+                   (3000 > time - g_Supervisor.startupTimeBeforeMenuMusic))
+            {
+                time = timeGetTime();
+            }
+            g_Supervisor.startupTimeBeforeMenuMusic = 0;
+            g_Supervisor.PlayAudio("bgm/th06_01.mid");
+        }
+        for (i = 0; i < 122; i++)
+        {
+            this->vm[i].pendingInterrupt = 1;
+            this->vm[i].flags |= 8;
```

Comment:
```suggestion
            this->vm[i].flags |= AnmVmFlags_8;
```

Hopefully one day we'll know what these do ^^